src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,96 @@
+package duke;
+
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.LocalStorage;
+import duke.storage.TaskList;
+import duke.ui.UI;
+
+/**
+ * duke.Duke Chat Bot!
+ */
+public class Duke {
+    private TaskList tasks;
+    private LocalStorage localStorage;
+
+    /**
+     * Constructor for duke.Duke
+     */
+    public Duke() {
+        this.tasks = new TaskList();
+    }
+
+    /**
+     * Constructor to instantiate duke.Duke bot.
+     * @param filePath path of local storage file.
+     */
+    public Duke(String filePath) {
+        TaskList tasks = new TaskList();
+        this.localStorage = new LocalStorage(filePath);
+        this.localStorage.loadTasks(tasks);
+        this.tasks = tasks;
+    }
+
+    /**
+     * Function to run duke.Duke bot.
+     */
+    public void run() {
+
+        UI.greet();
+
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        while (!input.equalsIgnoreCase("bye")) {
+            try {
+                Command com = new Parser(input, tasks).processRequest();
+                String response = com.execute(tasks);
+                UI.printRes(response);
+            } catch (DukeException error) {
+                UI.printRes(error.getMessage());
+            }
+            input = sc.nextLine();
+        }
+
+        localStorage.saveFile(tasks);
+        sc.close();
+
+        UI.exit();
+    }
+
+    public String getResponse(String request) {
+        try {
+            Command com = new Parser(request, tasks).processRequest();
+            String response = com.execute(tasks);
+            return response;
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+    /**
```

Comment:
Do you reckon there should be a new line in between the lines to separate the methods?